onnxruntime/test/providers/cpu/model_tests.cc

```diff
@@ -981,8 +1004,14 @@ ::std::vector<::std::basic_string<ORTCHAR_T>> GetParameterStrings() {
             return true;
           }
 #endif
+
           std::basic_string<PATH_CHAR_TYPE> p = ConcatPathComponent<PATH_CHAR_TYPE>(node_data_root_path, filename_str);
           std::basic_string<PATH_CHAR_TYPE> r = provider_name;
+          
+          if (CheckAndSkipTest(p, r)) {
```

Comment:
Then when it program is started, it needs to read all models. It could take more than hour an hour.

Reply:
What CheckAndSkipTest() does is the same as what previous [TEST_P(ModelTest, Run)](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/test/providers/cpu/model_tests.cc#L69) does.
They all need to read the information from all models, but is it going to take additional time of more than one hour?

From the CI log, I didn't see this happen.
Windows GPU CI:
This [PR](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=628395&view=results) -> 1 h 38 m
Other [PR](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=628325&view=results) -> 1 h 36 m
Linux GPU TensorRT CI:
This [PR](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=628399&view=results) -> 1 h 33 m
Other [PR](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=628399&view=results) -> 1 h 33 m

Reply:
"is it going to take additional time of more than one hour" It depends on if it was the first time the machine reads the data.

Reply:
Before my PR, if the machine loads the model data (happens TEST_P()) for the first time, it's going to take a long time as well.

Reply:
Anyway, I create a global map to hold OnnxModelInfo, so that later if the test needs model information, it can fetch from memory instead of from disk.

Reply:
The major difference is how long a developer needs to wait before get the first test running. Think if you are debugging a unit test failure. You use "--gtest_filter=xxx" to filter the tests so that it will only run the test you want to debug. But every time when you start the debugger, you need to wait a few minutes(or even half an hour) to reach your code. Would it be bad?

Reply:
> The major difference is how long a developer needs to wait before get the first test running. Think if you are debugging a unit test failure. You use "--gtest_filter=xxx" to filter the tests so that it will only run the test you want to debug. But every time when you start the debugger, you need to wait a few minutes(or even half an hour) to reach your code. Would it be bad?

Yes, that will be a problem if user only wants to debug a few tests. I'll see whether there is another better approach.

Reply:
We can leverage GTEST_SKIP()
https://github.com/google/googletest/blob/main/docs/advanced.md#skipping-test-execution